### DISCUSSION TITLE
How can I prevent `mockServiceWorker.js` from being included in my Vite prod build?

### DISCUSSION THREAD

**Discussion by jasikpark** [mswjs/msw#2210] [Q&A] [answered]
I'm using MSW with MirageJS in a Vite app, and `mockServiceWorker.js` is hosted from `/public`, which is an issue since it also gets copied into the production site. Is there a way to only add the file when in dev mode?

**Comment by dyljhd** [comment]
https://github.com/mswjs/msw/issues/291#issuecomment-2417014170

**Reply by jasikpark** [reply, marked as answer]
Found a simple solution for our purposes a bit back:

```
if (typeof process.env.VITE_MOCK_API !== 'undefined') {
    plugins.push(
      viteStaticCopy({
        targets: [
          {
            // Only serve MSW when API is mocked.
            src: 'test-setup/mockServiceWorker.js',
            dest: '/',
          },
        ],
      })
    );
  }
```

We just copy from a regular path into the dist only in dev.

**Reply by jasikpark** [reply]
Thx for the help 💜
